src/custom_nodes/east_ocr/east_ocr.cpp

```diff
@@ -0,0 +1,399 @@
+    int targetImageWidth = get_int_parameter("target_image_width", params, paramsCount, -1);
+    NODE_ASSERT(targetImageHeight > 0, "original image height must be larger than 0");
+    NODE_ASSERT(targetImageWidth > 0, "original image width must be larger than 0");
+    bool convertToGrayScale = get_string_parameter("convert_to_gray_scale", params, paramsCount) == "true";
+    float confidenceThreshold = get_float_parameter("confidence_threshold", params, paramsCount, -1.0);
+    NODE_ASSERT(confidenceThreshold >= 0 && confidenceThreshold <= 1.0, "confidence threshold must be in 0-1 range");
+    float overlapThreshold = get_float_parameter("overlap_threshold", params, paramsCount, 0.3);
+    NODE_ASSERT(overlapThreshold >= 0 && overlapThreshold <= 1.0, "non max suppression filtering overlap threshold must be in 0-1 range");
+    int maxOutputBatch = get_int_parameter("max_output_batch", params, paramsCount, 100);
+    NODE_ASSERT(maxOutputBatch > 0, "max output batch must be larger than 0");
+    bool debugMode = get_string_parameter("debug", params, paramsCount) == "true";
+    std::string imageLayout = get_string_parameter("image_layout", params, paramsCount, "NCHW");
+    NODE_ASSERT(imageLayout == "NCHW" || imageLayout == "NHWC", "image layout can be either NCHW or NHWC for both inputs and outputs");
+
+    const CustomNodeTensor* imageTensor = nullptr;
+    const CustomNodeTensor* scoresTensor = nullptr;
+    const CustomNodeTensor* geometryTensor = nullptr;
+
+    for (int i = 0; i < inputsCount; i++) {
+        if (std::strcmp(inputs[i].name, IMAGE_TENSOR_NAME) == 0) {
+            imageTensor = &(inputs[i]);
+        } else if (std::strcmp(inputs[i].name, SCORES_TENSOR_NAME) == 0) {
+            scoresTensor = &(inputs[i]);
+        } else if (std::strcmp(inputs[i].name, GEOMETRY_TENSOR_NAME) == 0) {
+            geometryTensor = &(inputs[i]);
+        } else {
+            std::cout << "Unrecognized input: " << inputs[i].name << std::endl;
+            return 1;
+        }
+    }
+
+    NODE_ASSERT(imageTensor->precision == FP32, "image input is not FP32");
+    NODE_ASSERT(scoresTensor->precision == FP32, "image input is not FP32");
+    NODE_ASSERT(geometryTensor->precision == FP32, "image input is not FP32");
+
+    NODE_ASSERT(imageTensor, "Missing image input");
+    NODE_ASSERT(scoresTensor, "Missing scores input");
+    NODE_ASSERT(geometryTensor, "Missing geometry input");
+
+    uint64_t imageHeight, imageWidth;
+    if (imageLayout == "NCHW") {
+        imageHeight = imageTensor->dims[2];
+        imageWidth = imageTensor->dims[3];
+    } else {
+        imageHeight = imageTensor->dims[1];
+        imageWidth = imageTensor->dims[2];
+    }
+
+    if (debugMode) {
+        std::cout << "Processing input tensor image resolution: " << cv::Size(imageHeight, imageWidth) << "; expected resolution: " << cv::Size(originalImageHeight, originalImageWidth) << std::endl;
```

Comment:
I don't recommend using cout in this way in customer facing example. Any concurrent execution of such node will result in lines broken in various way.

Use stringstream & just before exiting the custom node dump content of stringstream into cout.
While we loose getting those messages just as they occur we have readability for concurrent usage.

Reply:
String stream guard

Reply:
We agreed to not add guard into this example since it only prints in debug mode